src/common/post.ts

```diff
@@ -250,17 +250,32 @@ export type CreateSourcePostModeration = Omit<
   > & {
     contentHtml?: string;
     externalLink?: string;
+    postId?: string;
   };
 
 export const createSourcePostModeration = async (
   con: DataSource | EntityManager,
   args: CreateSourcePostModeration,
 ) => {
-  const newPost = con.getRepository(SourcePostModeration).create({
+  if (args.postId) {
+    const post = await con
+      .getRepository(Post)
+      .findOneByOrFail({ id: args.postId });
+
+    const isUnauthorizedEdit =
+      (post.type === PostType.Freeform && args.createdById !== post.authorId) ||
+      (post.type !== PostType.Freeform && args.createdById !== post.scoutId);
```

Comment:
Why do we need `post.type` check here?

Reply:
@capJavert This was the reason I asked you about yesterday who is the post creator. I understood you such as that when the post is scraped or written on our site, the "creator" is the author id, and for share/link it is scout id. 

Did I understand that correctly?

Reply:
We should just check whether the user is the `authorId`. We only allow editing on `freeform` anyway. Also, the scout part is for community picks. Even `share` post can have the right `authorId`.

Reply:
Talked with Lee in DMs and he clarified that we do not use scoutId in squad, and both share and freeform use authorId for the "creator".

WIll therefor only check if createdById === authorId